test/framework/test_db_manager.cpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "framework/test_db_manager.hpp"
+
+#include "ametsuchi/impl/k_times_reconnection_strategy.hpp"
+#include "ametsuchi/impl/pool_wrapper.hpp"
+#include "ametsuchi/impl/postgres_options.hpp"
+#include "framework/config_helper.hpp"
+#include "framework/result_gtest_checkers.hpp"
+#include "logger/logger_manager.hpp"
+#include "main/impl/pg_connection_init.hpp"
+
+using namespace framework::expected;
+using namespace integration_framework;
+using namespace iroha::ametsuchi;
+using namespace iroha::expected;
+using namespace iroha::integration_framework;
+
+static constexpr size_t kMaxRandomDbNameAttempts = 8;
+
+/// Drops a database on destruction.
+class TestDbManager::DbDropper {
+ public:
+  DbDropper(std::unique_ptr<soci::session> management_session,
+            std::string dropped_db_name)
+      : management_session_(std::move(management_session)),
+        dropped_db_name_(std::move(dropped_db_name)) {}
+  ~DbDropper() {
+    *management_session_ << "DROP DATABASE " + dropped_db_name_;
+  }
+ private:
+  std::unique_ptr<soci::session> management_session_;
+  std::string dropped_db_name_;
+};
+
+Result<std::unique_ptr<TestDbManager>, std::string>
+TestDbManager::createWithRandomDbName(
+    size_t sessions, logger::LoggerManagerTreePtr log_manager) {
+  size_t random_db_name_attempts = 0;
+  static const auto default_creds = getPostgresCredsOrDefault();
+  while (random_db_name_attempts++ < kMaxRandomDbNameAttempts) {
+    auto pg_opts = std::make_unique<PostgresOptions>(
+        default_creds,
+        getRandomDbName(),
+        log_manager->getChild("PostgresOptions")->getLogger());
+    auto db_exists_result =
+        PgConnectionInit::checkIfWorkingDatabaseIfExists(*pg_opts);
+    if (auto e = resultToOptionalError(db_exists_result)) {
+      return std::move(e).value();
+    }
+    const bool db_exists = resultToOptionalValue(db_exists_result).value();
+    if (not db_exists) {
+      return PgConnectionInit::createDatabaseIfNotExist(*pg_opts) |
+          [&](bool db_was_created) {
+            EXPECT_TRUE(db_was_created);
+            return PgConnectionInit::prepareConnectionPool(
+                KTimesReconnectionStrategyFactory{10},
```

Comment:
Without going down through the code it is not clear what constant 10 means.

Could you please comment or name it?